sycl/test-e2e/Basic/event_creation.cpp

```diff
@@ -17,7 +17,7 @@ int main() {
   try {
     std::cout << "Create default event" << std::endl;
     sycl::event e;
-  } catch (sycl::device_error e) {
+  } catch (const sycl::exception &e) {
```

Comment:
Should we also check error code here now?

Reply:
I'd say this test is useless, and I can't understand its intent in regards to the `catch` even at oldest point I can find (https://github.com/intel/llvm/blob/5ee43c562a4865457a2dc2d6af7da81628d33386/SYCL/Basic/event_creation.cpp).

As such, I don't think we need to check error code here. I'm thinking about just removing it in a separate PR.